fusion-cli/build/plugins/instrumented-import-dependency-template-plugin.js

```diff
@@ -9,13 +9,84 @@
 /* eslint-env node */
 
 /*::
-import type {ClientChunkMetadataState, ClientChunkMetadata} from "../types.js";
+import type {
+  ClientChunkMetadataState,
+  ClientChunkMetadata,
+  TranslationsManifest,
+} from "../types.js";
+
+type InstrumentationPluginOpts =
+  | ClientPluginOpts
+  | ServerPluginOpts;
+
+type ServerPluginOpts = {
+  compilation: "server",
+  clientChunkMetadata: ClientChunkMetadataState
+};
+
+type ClientPluginOpts = {
+  compilation: "client",
+  i18nManifest: TranslationsManifest
+};
 */
 
 const ImportDependency = require('webpack/lib/dependencies/ImportDependency');
 const ImportDependencyTemplate = require('webpack/lib/dependencies/ImportDependency')
   .Template;
 
+class InstrumentedImportDependency extends ImportDependency {
+  constructor(dep, opts) {
+    super(dep.request, dep.originModule, dep.block);
+    this.module = dep.module;
+    this.loc = dep.loc;
+
+    if (opts.compilation === 'client') {
+      this.translationsManifest = opts.i18nManifest;
+    }
+    if (opts.compilation === 'server' && opts.clientChunkMetadata) {
+      // server
+      opts.clientChunkMetadata.result.then(chunkIndex => {
+        this.clientChunkIndex = chunkIndex.fileManifest;
+      });
+    }
+  }
+  getInstrumentation() {
+    if (this.clientChunkIndex) {
+      // server-side, use values from client bundle
+      let ids = this.clientChunkIndex.get(
+        (this.module && this.module.resource) || this.originModule.resource
+      );
+      this.chunkIds = ids ? Array.from(ids) : [];
+    } else {
+      // client-side, use built-in values
+      this.chunkIds = getChunkGroupIds(this.block.chunkGroup);
+    }
+
+    this.translationKeys = [];
+    if (this.translationsManifest) {
+      const modules = getChunkGroupModules(this);
+      for (const module of modules) {
+        if (this.translationsManifest.has(module)) {
+          const keys = this.translationsManifest.get(module).keys();
+          this.translationKeys.push(...keys);
+        }
+      }
+    }
+
+    return {
+      chunkIds: this.chunkIds,
+      translationKeys: this.translationKeys,
+    };
+  }
+  updateHash(hash) {
+    super.updateHash(hash);
+    const {translationKeys} = this.getInstrumentation();
+    // Invalidate this dependency when the translation keys change
+    // Necessary for HMR
+    hash.update(JSON.stringify(translationKeys));
```

Comment:
Dependencies only depend on module id by default. This logic will tell the dependency to update when `translationKeys` changes.